File: frontend/hostel-frontend/src/services/complaintService.js
import API from "../api/axios";

/* STUDENT */

export const raiseComplaint = (complaint) => {
  return API.post("/complaints/raise", complaint);
};

export const getStudentComplaints = (studentId) => {
  return API.get(`/complaints/student/${studentId}`);
};


/* ADMIN */

export const getAllComplaints = () => {
  return API.get("/complaints");
};

export const updateComplaintStatus = (complaintId, status) => {
  return API.put(`/complaints/${complaintId}/status?status=${status}`);
};

File: backend/hostel-management/src/main/java/com/hostel/hostel_management/service/ComplaintService.java
package com.hostel.hostel_management.service;

import java.util.List;

import com.hostel.hostel_management.entity.Complaint;
import com.hostel.hostel_management.entity.enums.ComplaintStatus;

public interface ComplaintService {
    Complaint raiseComplaint(Complaint complaint);

    List<Complaint> getStudentComplaints(Long studentId);
    Complaint updateComplaintStatus(Long complaintId, ComplaintStatus status);

    List<Complaint> getAllComplaints();
}
